src/main/java/uk/gov/ch/controller/update/OverseasEntityDataController.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.ch.controller.update;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.ch.OracleQueryApplication;
+import uk.gov.ch.exception.OverseasEntityEmailAddressNotFoundException;
+import uk.gov.ch.model.update.OverseasEntityDataJson;
+import uk.gov.ch.service.update.impl.OverseasOverseasEntityDataServiceImpl;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.logging.util.DataMap;
+
+import javax.validation.constraints.Pattern;
+
+@RestController
+@Validated
+public class OverseasEntityDataController {
+    @Autowired
+    private OverseasOverseasEntityDataServiceImpl entityDataService;
+
+    private static final Logger logger = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    @GetMapping("/overseas-entity/{companyNumber}/entity-data")
+    public ResponseEntity<OverseasEntityDataJson> getEntityEmail(@PathVariable("companyNumber")
+                                                                 @Pattern(regexp = "^OE\\d{6}$", message = "Invalid overseas entity number") String companyNumber) {
+
+        DataMap dataMap = new DataMap.Builder(null).companyNumber(companyNumber).build();
```

Comment:
The lack of a no arg constructor has been raised with the developer who wrote the new builder class. We'll change to use it once available. Setting the company name arg should be fine as there is a test to not add nulls to the map.

Reply:
Note that a PR is now up for this: https://github.com/companieshouse/structured-logging/pull/72.

I think it will be merged soon.